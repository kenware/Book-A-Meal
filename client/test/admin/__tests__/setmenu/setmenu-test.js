import React from 'react';
import { shallow } from 'enzyme';
import { SetMenu, mapStateToProps, mapDispatchToProps } from '../../../../components/admin/setMenu';
import { props, emptyProps } from '../../adminMockData';

describe('Test Component of SETMENU component', () => {
  it('renders correctly', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should have one order-container class', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('.order-container')).toHaveLength(1);
  });

  it('should display one order-wrapper', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('.order-wrapper')).toHaveLength(1);
  });
  it('should have form-fields', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('.form-field')).toHaveLength(2);
  });

  it('should have one table', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('table')).toHaveLength(1);
  });

  it('should have only 3 tr on the table', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('tr')).toHaveLength(3);
  });

  it('should have only 4 td on the table', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('td')).toHaveLength(6);
  });

  it('should have only 4 th on the table', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('th')).toHaveLength(3);
  });

  it('should display two menu meals', () => {
    const tree = shallow(<SetMenu {...props} />);
    expect(tree.find('.tr')).toHaveLength(2);
  });

  it('should display no menu meals with empty meal props', () => {
    const tree = shallow(<SetMenu {...emptyProps} />);
    expect(tree.find('.tr')).toHaveLength(0);
  });

  it('should respond to change event and change the state of the title input and expire time and add meal to the menu', () => {
    const tree = shallow(<SetMenu {...props} />);
    // onChange event
    tree.find('#title').simulate('change', { target: { name: 'title', value: 'Today menu' } });
    expect(tree.state('title')).toEqual('Today menu');
    // onChange event
    tree.find('#expire').simulate('change', { target: { name: 'orderBefore', value: '10' } });
    expect(tree.state('orderBefore')).toEqual('10');
    tree.find('#rice').simulate('click');
    const wrapper = tree.instance();
    wrapper.addMenu();
    expect(tree).toMatchSnapshot();
  });
  it('should respond to change event and change the state of the title input and return error while addMeal method when orderBefore field is not supplied', () => {
    const tree = shallow(<SetMenu {...props} />);
    // onChange event
    tree.find('#title').simulate('change', { target: { name: 'title', value: 'Today menu' } });
    expect(tree.state('title')).toEqual('Today menu');
    tree.find('#rice').simulate('click');
    const wrapper = tree.instance();
    wrapper.addMenu();
    expect(tree).toMatchSnapshot();
  });

  it('should respond to a click event by dispalying a modal when caterer want to set menu by selecting a meal', () => {
    const tree = shallow(<SetMenu {...props} />);
    // onClick event
    tree.find('#rice').simulate('click');
    expect(tree).toMatchSnapshot();
  });

  it('should call addMenu method onclick event', () => {
    const addmenu = jest.spyOn(SetMenu.prototype, 'addMenu');
    const tree = shallow(<SetMenu {...props} />);
    // onClick event
    tree.find('#rice').simulate('click');
    tree.find('#addMenu').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(addmenu).toHaveBeenCalled();
  });

  it('should cancel addMenu modal cancelAdd method onclick event', () => {
    const removeMenu = jest.spyOn(SetMenu.prototype, 'cancelAdd');
    const tree = shallow(<SetMenu {...props} />);
    // onClick event
    tree.find('#rice').simulate('click');
    tree.find('#removeMenu').simulate('click');
    expect(tree).toMatchSnapshot();
    expect(removeMenu).toHaveBeenCalled();
  });
  it('should should respond to cancelAdd method that close modal', () => {
    const tree = shallow(<SetMenu {...props} />);
    // onClick event
    const wrapper = tree.instance();
    wrapper.cancelAdd();
    expect(tree).toMatchSnapshot();
  });
  it('should should respond to lifeCycle Methods', () => {
    const tree = shallow(<SetMenu {...props} />);
    // onClick event
    tree.instance().constructor.getDerivedStateFromProps({ errorMessage: { setMenuError: '' }, successMessage: { setMenuSuccess: '' }, actions: { clearMessages: jest.fn() } });
    expect(tree).toMatchSnapshot();
    tree.instance().constructor.getDerivedStateFromProps({ errorMessage: { setMenuError: 'error' }, successMessage: { setMenuSuccess: 'success' }, actions: { clearMessages: jest.fn() } });
    expect(tree).toMatchSnapshot();
  });
  it('should respond to mapStateToProps methods', () => {
    const ownProps = { match: { params: { mealId: 1 } } };
    const tree = mapStateToProps(props, ownProps);
    expect(tree).toMatchSnapshot();
  });
  it('should respond to mapDispatchToProps methods', () => {
    const tree = mapDispatchToProps(emptyProps);
    expect(tree).toMatchSnapshot();
  });
});
