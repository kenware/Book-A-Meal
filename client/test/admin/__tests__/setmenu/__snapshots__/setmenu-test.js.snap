// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component of SETMENU component renders correctly 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should call addMenu method onclick event 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      >
        Title is required
      </span>
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should cancel addMenu modal cancelAdd method onclick event 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      >
        Title is required
      </span>
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should respond to a click event by dispalying a modal when caterer want to set menu by selecting a meal 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      >
        Title is required
      </span>
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should respond to change event and change the state of the title input and expire time and add meal to the menu 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order "
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
        rice
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should respond to change event and change the state of the title input and return error while addMeal method when orderBefore field is not supplied 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      >
        Order Expiring time is required
      </span>
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should respond to mapDispatchToProps methods 1`] = `
Object {
  "actions": Object {
    "clearMessages": [Function],
    "getNotifications": [Function],
    "getUser": [Function],
    "loadErrorMessage": [Function],
    "loadMostOrdered": [Function],
    "loadMostOrderedMeal": [Function],
    "loadNotifications": [Function],
    "loadSuccessMessage": [Function],
    "loadUser": [Function],
    "login": [Function],
    "refreshToken": [Function],
    "register": [Function],
    "resetLink": [Function],
    "updateProfile": [Function],
    "upgrade": [Function],
  },
  "menuActions": Object {
    "getMenu": [Function],
    "loadErrorMessage": [Function],
    "loadSuccessMessage": [Function],
    "loadTodayMenu": [Function],
    "setMenu": [Function],
  },
}
`;

exports[`Test Component of SETMENU component should respond to mapStateToProps methods 1`] = `
Object {
  "errorMessage": Object {
    "createMealError": "Wrong credentials",
  },
  "meals": Array [
    Object {
      "descrption": "Very delicious",
      "id": 1,
      "name": "rice",
      "price": "200",
    },
    Object {
      "descrption": "Very delicious",
      "id": 2,
      "name": "beans",
      "price": "200",
    },
  ],
  "successMessage": Object {
    "createMealSuccess": "meal created",
  },
}
`;

exports[`Test Component of SETMENU component should should respond to cancelAdd method that close modal 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should should respond to lifeCycle Methods 1`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Component of SETMENU component should should respond to lifeCycle Methods 2`] = `
<div
  className="order-wrapper order-container"
>
  <div
    className="modal-order modal"
  >
    <button
      className="remove-modal"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "red",
          "display": "block",
          "float": "right",
        }
      }
    >
      ×
    </button>
    <div
      className="modal-order-content"
      style={
        Object {
          "margin": "1rem 1rem 1rem 1rem",
        }
      }
    >
      <p
        className="justify l-r-pad-text"
      >
         Add 
         to todays menu
      </p>
    </div>
    <div
      className="modal-order-content"
    >
      <button
        className="remove-modal"
        id="addMenu"
        onClick={[Function]}
      >
        Add
      </button>
      <button
        className="remove-modal"
        id="removeMenu"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
  <h2
    style={
      Object {
        "marginTop": "2rem",
      }
    }
  >
    SET MENU FOR THE DAY
  </h2>
  <h4
    className="p-color text-center"
  >
    Enter Menu Title and the Expire Time(24 Hours Format) in Hours
  </h4>
  <h5
    className="p-color text-center"
  >
    Eg Expire time of 17 
    <em
      className="fa fa-long-arrow-right"
    />
     5pm
  </h5>
  <h4
    className="danger text-center"
  />
  <h4
    className="y-color text-center"
  />
  <div
    className="form-field"
  >
    <label
      htmlFor="name"
    >
      Title 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Keny"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-field"
  >
    <label
      htmlFor="password"
    >
      Expire Time 
      <br />
      <span
        className="text-center danger"
      />
    </label>
    <input
      id="expire"
      name="orderBefore"
      onChange={[Function]}
      required={true}
      type="number"
    />
  </div>
  <h4
    className="p-color text-center"
  >
    Set menu from the meal options
  </h4>
  <table>
    <tbody>
      <tr>
        <th>
          id
        </th>
        <th>
          Name
        </th>
        <th>
          Price ($)
        </th>
        <th>
          Add to Menu
        </th>
      </tr>
      <tr
        className="tr"
        key="1"
      >
        <td>
          1
        </td>
        <td>
          rice
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="rice"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
      <tr
        className="tr"
        key="2"
      >
        <td>
          2
        </td>
        <td>
          beans
        </td>
        <td>
          200
        </td>
        <td>
          <button
            id="beans"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-list-alt y-color"
            />
             Add
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
